docs/workflow/testing/libraries/testing-android.md

```diff
@@ -0,0 +1,57 @@
+# Testing Libraries on Android
+
+To build the tests and run them on Android (devices or emulators) you need the following prerequisites.
+
+- [Android NDK](https://developer.android.com/ndk/downloads)
+- [Android SDK](https://developer.android.com/studio)
+- OpenJDK
+
+(TODO: provide command-line steps to download them with dependencies)
+
+Once SDKs are downloaded, set `ANDROID_NDK_ROOT`(CLARIFY: looks like `ANDROID_NDK_HOME` should also be set) and `ANDROID_SDK_ROOT`.
+Example:
+```
+export ANDROID_SDK_ROOT=/Users/egorbo/Library/Android/sdk
+export ANDROID_NDK_ROOT=/Users/egorbo/android-ndk-r21b
+export ANDROID_NDK_HOME=$ANDROID_NDK_ROOT
+```
+
+Next, we need OpenSSL binaries and headers, we haven't properly integrated this dependency yet (it's an ongoing discussion) but there is a workaround:
+
+- Download and unzip https://maven.google.com/com/android/ndk/thirdparty/openssl/1.1.1g-alpha-1/openssl-1.1.1g-alpha-1.aar
+- Set these env variables:
+```
+GOOGLE_OPENSSL=/Users/egorbo/prj/openssl-1.1.1g-alpha-1.aar/prefab/modules
+export AndroidOpenSslHeaders="$GOOGLE_OPENSSL/ssl/include"
+export AndroidOpenSslCryptoLib="$GOOGLE_OPENSSL/crypto/libs/android.x86_64/libcrypto.so"
+export AndroidOpenSslLib="$GOOGLE_OPENSSL/ssl/libs/android.x86_64/libssl.so"
+```
+**IMPORTANT:** make sure correct ABIs are used in the path, e.g. `-arch x64` -> `android.x86_64` (TODO: auto-detect)
```

Comment:
maybe we should simplify this by only needing e.g. GOOGLE_OPENSSL and letting the build figure out the right paths?

Reply:
Yes, I'll do it in a separate PR to keep this docs-only